File: advanced-programming-methods/labs/toy-interpreter-sharp/model/statement/ForkStatement.cs
using System;

namespace Model
{
    [Serializable()]
    public class ForkStatement : Statement
    {
        private Statement program;

        public ForkStatement(Statement program)
        {
            this.program = program;
        }

        public Statement GetProgram()
        {
            return program;
        }

        /// <inheritdoc />
        public override string ToString()
        {
            return string.Format("fork({0})", program.ToString());
        }

        /// <inheritdoc />
        public override Statement CloneDeep()
        {
            return new ForkStatement(program.CloneDeep());
        }
    }
}

File: advanced-programming-methods/labs/toy-interpreter-java/src/model/statement/ForkStatement.java
package model.statement;

public class ForkStatement extends Statement
{
    private Statement program;

    public ForkStatement(Statement program)
    {
        this.program = program;
    }

    /**
     * Getter for property 'program'.
     *
     * @return Value for property 'program'.
     */
    public Statement getProgram()
    {
        return program;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String toString()
    {
        return String.format("fork(%s)", program.toString());
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public Statement cloneDeep()
    {
        return new ForkStatement(program.cloneDeep());
    }
}
